diagnostics/collect-wsl-logs.ps1

```diff
@@ -130,6 +130,104 @@ if ($Dump)
     }
 }
 
+# Collect networking state relevant for WSL
+# Using a try/catch for commands below, as some of them do not exist on all OS versions
+
+Write-Host "`nCollecting additional network state..."
+
+$networkingFolder = "$folder/networking"
+mkdir -p $networkingFolder
+
+# Host networking info
+try
+{
+    Get-NetAdapter -includeHidden | select Name,ifIndex,NetLuid,InterfaceGuid,Status,MacAddress,MtuSize,InterfaceType,Hidden,HardwareInterface,ConnectorPresent,MediaType,PhysicalMediaType | Out-File -FilePath "$networkingFolder/Get-NetAdapter.log" -Append
+}
+catch {}
+
+try
+{
+    Get-NetIPConfiguration -All -Detailed | Out-File -FilePath "$networkingFolder/Get-NetIPConfiguration.log" -Append
+}
+catch {}
+
+try
+{
+    Get-NetFirewallHyperVVMCreator | Out-File -FilePath "$networkingFolder/Get-NetFirewallHyperVVMCreator.log" -Append
+}
+catch {}
+
+try
+{
+    Get-NetFirewallHyperVVMSetting -PolicyStore ActiveStore | Out-File -FilePath "$networkingFolder/Get-NetFirewallHyperVVMSetting_ActiveStore.log" -Append
+}
+catch {}
+
+try
+{
+    Get-NetFirewallHyperVProfile -PolicyStore ActiveStore | Out-File -FilePath "$networkingFolder/Get-NetFirewallHyperVProfile_ActiveStore.log" -Append
+}
+catch {}
+
+try
+{
+    Get-NetFirewallHyperVPort | Out-File -FilePath "$networkingFolder/Get-NetFirewallHyperVPort.log" -Append
+}
+catch {}
+
+try
+{
+    & hnsdiag.exe list all 2>&1 > $networkingFolder/hnsdiag_list_all.log
+}
+catch {}
+
+try
+{
+    & hnsdiag.exe list endpoints -df 2>&1 > $networkingFolder/hnsdiag_list_endpoints.log
+}
+catch {}
+
+try
+{
+    foreach ($port in Get-NetFirewallHyperVPort)
+    {
+		& vfpctrl.exe /port $port.PortName /get-port-state 2>&1 > "$networkingFolder/vfp-port-$($port.PortName)-get-port-state.log"
+	    & vfpctrl.exe /port $port.PortName /list-rule 2>&1 > "$networkingFolder/vfp-port-$($port.PortName)-list-rule.log"
+    }
+}
+catch {}
+
+try
+{
+    & vfpctrl.exe /list-vmswitch-port 2>&1 > $networkingFolder/vfpctrl_list_vmswitch_port.log
+}
+catch {}
+
+# Container networking info
+try
+{
+	wsl -e ip a s | Out-File -FilePath "$networkingFolder/wsl_ip_a_s.log" -Append
```

Comment:
We interact with the user distribution as part of the log collection because this might block forever is WSL is in a bad state. 

If we need container information, that should be done in a separate script that can users can run if needed

Reply:
Could an alternative solution be to run the WSL commands asynchronously with some kind of timeout? 
maybe something using https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/wait-job?view=powershell-7.3 ? (still need to look into how to use this exactly)

Reply:
I would prefer to avoid doing this since we don't know which version of powershell the user will have installed and we also don't know which distribution the issue is with (calling wsl.exe will invoke the default distribution, but that might not the one having the issue).

We have [dedicated script for that purpose](https://github.com/microsoft/WSL/blob/master/diagnostics/networking.bat) that we ask users to run specifically for networking issues. I'd recommend adding to that one instead

Reply:
Makes sense, thanks Pierre, will add to that script then.